src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -169,18 +170,27 @@ private function dumpArrayStructure(string $output, string $input, StructureShap
     private function dumpArrayList(string $output, string $input, string $contextProperty, ListShape $shape): string
     {
         $memberShape = $shape->getMember()->getShape();
+        static $counter = -1;
 
-        return strtr('
-            $index = -1;
-            foreach (INPUT as $listValue) {
-                $index++;
-                MEMBER_CODE
-            }
-        ',
-            [
-                'INPUT' => $input,
-                'MEMBER_CODE' => $memberCode = $this->dumpArrayElement(sprintf('%s[$index]', $output), '$listValue', $contextProperty, $memberShape, true),
-            ]);
+        try {
+            ++$counter;
+            $cleanCounter = $counter ?: '';
+
+            return strtr('
+                $indexCOUNTER = -1;
+                foreach (INPUT as $listValueCOUNTER) {
+                    $indexCOUNTER++;
```

Comment:
Only in `BatchExecuteStatementRequest` this will be `$index1`. Do we really need COUNTER?

Reply:
Oh, so we have `$index` and `$index1` if we got a second one.. Hm.. Okey. This if fine then.

Reply:
yes, with recursion, we have several counters